spokestack/asr/cloud_client.py

```diff
@@ -0,0 +1,106 @@
+"""
+This module contains the websocket logic used to communicate with
+Spokestack's cloud-based ASR service.
+"""
+import base64
+import hashlib
+import hmac
+import json
+
+from websocket import WebSocket  # type: ignore
+
+
+class CloudClient:
+    """ spokestack cloud client """
+
+    def __init__(
+        self,
+        socket_url: str,
+        key_id: str,
+        key_secret: str,
+        audio_format: str = "PCM16LE",
+        sample_rate: int = 16000,
+        language: str = "en",
+        limit: int = 10,
+    ) -> None:
+        self._body: str = json.dumps(
+            {
+                "format": audio_format,
+                "rate": sample_rate,
+                "language": language,
+                "limit": limit,
+            }
+        )
+
+        self._socket_url: str = socket_url
+        self._key_id: str = key_id
+        self._key: bytes = key_secret.encode("utf-8")
+        signature = hmac.new(
```

Comment:
this is fine for now and I know the android version does this, but I don't really like pre-calculating the request signature like this
at some point, we will add things like timestamps for replay attacks, and that will require rewriting all of this to calculate the request signature correctly before it is submitted